internal/workflow_testsuite.go

```diff
@@ -190,6 +190,32 @@ func (t *TestWorkflowEnvironment) OnWorkflow(workflow interface{}, args ...inter
 	return t.wrapCall(call)
 }
 
+const mockMethodForSignalExternalWorkflow = "workflow.SignalExternalWorkflow"
+
+// OnSignalExternalWorkflow setup a mock for sending signal to external workflow.
+// This TestWorkflowEnvironment handles sending signals between the workflows that are started from the root workflow.
+// For example, sending signals between parent and child workflows. Or sending signals between 2 child workflows.
+// However, it does not know what to do if your tested workflow code is sending signal to external unknown workflows.
+// In that case, you will need to setup mock for those signal calls.
+// Some examples of how to setup mock:
+//
+// * mock for specific target workflow that matches specific signal name and signal data
+// 	 env.OnSignalExternalWorkflow("test-domain", "test-workflow-id1", "test-runid1", "test-signal", "test-data").Return(nil).Once()
+// * mock for anything and succeed the send
+// 	 env.OnSignalExternalWorkflow(mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+// * mock for anything and fail the send
+// 	 env.OnSignalExternalWorkflow(mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(errors.New("unknown external workflow")).Once()
```

Comment:
Does it convert to an error that Cadence actually returns from Signal?

Reply:
no, it directly returns what ever mock supplied. We don't have an error type defined for signal error. In production code we use fmt.Errorf(...) to create error for signal failure.